mapactionpy_arcmap/map_chef.py

```diff
@@ -14,22 +17,46 @@ class MapChef:
     Worker which creates a Map based on a predefined "recipe" from a cookbook
     """
 
-    def __init__(self, mxd, cookbookJsonFile, layerPropertiesJsonFile, crashMoveFolder, layerDirectory):
+    def __init__(self,
```

Comment:
The number of parameters can be reduced by just passing the `cmf` and `event` objects as parameters. This would cut out the need for `cookbookJsonFile`, `layerPropertiesJsonFile`,
`crashMoveFolder` and `layerDirectory` parameters.

Futhermore a lot of the MapChef instance and local variables are repeating information already encapsulated in the `cmf` and `event` objects. 

Hence if MapChef hence instance variables of the `cmf` and `event` object themselves, quite I think quite a lot of the "noise" of parameter handling and validation can pared back. There is a lot of checking the existence of particular files and directories. If you don't trust the parameter passed to MapChef you can call `self.cmf.verify_paths()` within the constructor and then trust it from there onwards.

I've annotated a version on the constructor below to try and show what I mean. I've not followed through the consequences for the rest of the file. There are 16 instances of `os.path` in map_chef.py. I think most of these could be removed along with several of the nested loops and if statements within the `processLayer()` method.

```
    def __init__(self,
                 mxd,
                 cmf,
                 event,
                 versionNumber=1):
        """
        Arguments:
           mxd {MXD file} -- MXD file.
           crashMoveFolder {CrashMoveFolder} -- A CrashMoveFolder object.
		   event {Event} -- An Event object.
           versionNumber {int} -- version number of map.
        """
        self.mxd = mxd
	self.cmf = cmf
	self.event = event

	# If you're still feeling paranoid you could validate the cmf object if you don't 
	# trust what you've been passed.
	self.cmf.verify_paths()

        # not required self.layerPropertiesJsonFile = layerPropertiesJsonFile
	# now accessible as self.cmf.layer_properties
		
        # not required  self.cookbookJsonFile = cookbookJsonFile
	# now accessible self.cmf.map_definitions
        
	# not required self.crashMoveFolder = crashMoveFolder
	# only used to resolve relative paths which are handled by the cmf object
		
        # not required self.layerDirectory = layerDirectory
	# now accessible self.cmf.layer_rendering
		
        self.cookbook = MapCookbook(self.cmf.map_definitions)
        self.layerProperties = LayerProperties(self.cmf.layer_properties)
        self.legendEntriesToRemove = list()

	# No need to duplicate these objects:
        # eventFilePath = os.path.join(crashMoveFolder, "event_description.json")
        # cmfFilePath = os.path.join(crashMoveFolder, "cmf_description.json")

	# No need for the initial assignment to None 
        # self.event = None
        # self.cmfConfig = None
        # self.namingConvention = None

        self.summary = "Insert summary here"
        self.dataSources = set()
        self.versionNumber = versionNumber

	# No need to duplicate the CMF and Event objects.
	# See below for the NamingConvention
        # if os.path.exists(eventFilePath):
        #     self.event = Event(eventFilePath)
        # if os.path.exists(cmfFilePath):
        #     self.cmfConfig = CrashMoveFolder(cmfFilePath)
        #    self.namingConvention = NamingConvention(os.path.join(crashMoveFolder, self.cmfConfig.dnc_definition))

	# The NamingConvention class is generic, and is only made specfic based on the definition file passed to it's
	# constructor. The are already definition files for the Data Naming Convention, Layer Naming Convention and 
	# MXD Naming Convention. Therefore when initialising a NamingConvention object, choice a variable name which
	# describes _which_ naming convention it refers to.
	# Also `os.path.join` not required here.
	self.data_naming_convention = NamingConvention(self.cmfConfig.dnc_definition)
```